patches/web3-eth-contract+1.2.6.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/node_modules/web3-eth-contract/types/index.d.ts b/node_modules/web3-eth-contract/types/index.d.ts
+index 6f141ab..1207009 100644
+--- a/node_modules/web3-eth-contract/types/index.d.ts
++++ b/node_modules/web3-eth-contract/types/index.d.ts
+@@ -78,7 +78,7 @@ export interface Options extends ContractOptions {
+ }
+ 
+ export interface DeployOptions {
+-    data: string;
```

Comment:
I now use 'patches' folder as a ad-hoc list of pull requests I want to submit once I have time. 
Also, I would need to do this weird 'null as string' on every call to '.deploy', which is not pretty.

Reply:
the point is that deploy requires data - its not optional, so changing the library (instead of changing your code) is not the right solution.
 its your code that says "I know my var is nullable, but I also know at this point that I checked for null"
you could actually wrap with `if data!=null ()` and it will recognize it.

Reply:
It is not that.
Data parameter is optional.
If you construct the contract object with a bytecode and ABI